packages/sql-pglite/src/PgLiteClient.ts

```diff
@@ -0,0 +1,343 @@
+  readonly transformQueryNames?: ((str: string) => string) | undefined
+  readonly transformJson?: boolean | undefined
+  readonly applicationName?: string | undefined
+  readonly spanAttributes?: Record<string, unknown> | undefined
+}
+
+/**
+ * @category constructors
+ * @since 1.0.0
+ */
+export const make = <TExtensions extends Extensions = Extensions>(
+  options: PgliteClientConfig<TExtensions>
+): Effect.Effect<PgliteClient<Extensions>, SqlError, Scope.Scope | Reactivity.Reactivity> =>
+  Effect.gen(function*() {
+    const compiler = makeCompiler(options.transformQueryNames, options.transformJson)
+    const transformRows = options.transformResultNames
+      ? Statement.defaultTransforms(options.transformResultNames, options.transformJson).array
+      : undefined
+
+    const client: PGlite = yield* Effect.tryPromise({
+      try: () =>
+        PGlite.create(
+          options.dataDir || "",
+          options
+        ),
+      catch: (cause) => new SqlError({ cause, message: "PgliteClient: Failed to connect" })
+    })
+
+    yield* Effect.tryPromise({
+      try: () => client.query("SELECT 1"),
+      catch: (cause) => new SqlError({ cause, message: "PgliteClient: Failed to query" })
+    })
+
+    class ConnectionImpl implements Connection {
+      constructor(private readonly pg: PGlite) {}
+
+      private run(query: Promise<any>) {
+        return Effect.async<ReadonlyArray<any>, SqlError>((resume) => {
+          query.then(
+            (result) => {
+              resume(Effect.succeed(result.rows))
+            },
+            (cause) => {
+              resume(new SqlError({ cause, message: "Failed to execute statement" }))
+            }
+          )
+          // PGlite doesn't have a cancel method like postgres.js
+          return Effect.succeed(void 0)
+        })
+      }
+
+      execute(
+        sql: string,
+        params: ReadonlyArray<Primitive>,
+        transformRows?: (<A extends object>(row: ReadonlyArray<A>) => ReadonlyArray<A>) | undefined,
+        unprepared?: boolean
+      ) {
+        return transformRows
+          ? Effect.map(
+            this.run(
+              unprepared ? this.pg.exec(sql, params as any) : this.pg.query(sql, params as any)
+            ),
+            transformRows
+          )
+          : unprepared
+          ? this.run(this.pg.exec(sql, params as any))
+          : this.run(this.pg.query(sql, params as any))
+      }
+      executeRaw(sql: string, params: ReadonlyArray<Primitive>) {
+        return this.run(this.pg.exec(sql, params as any))
+      }
+      executeWithoutTransform(sql: string, params: ReadonlyArray<Primitive>) {
+        return this.run(this.pg.query(sql, params as any))
+      }
+      executeValues(sql: string, params: ReadonlyArray<Primitive>) {
+        // PGlite doesn't have a values() method like postgres.js
+        // We'll just return the regular query results
+        return this.execute(sql, params, (r) => Object.values(r))
```

Comment:
I think you will need an .map here? The results will be an array?

Reply:
Maybe I was confused about what executeValues should return. Is it supposed to be an array of arrays of values? That's what this should do since the function param is the row transform, it transforms each row object into an array of values.